CRM/Xcm/Matcher/AddressMatcher.php

```diff
@@ -59,6 +66,22 @@ public function matchContact(&$contact_data, $params = NULL) {
       'postal_code'    => $contact_data['postal_code'],
       'city'           => $contact_data['city'],
       );
+    if ($this->isStreetAddressParsingEnabled) {
+      if (empty($address_query['street_address'])) {
+        unset($address_query['street_address']);
+      }
+      if (array_key_exists('street_name', $contact_data)) {
+        $address_query['street_name'] = $contact_data['street_name'];
+      }
+      if (array_key_exists('street_number', $contact_data)) {
+        $address_query['street_number'] = $contact_data['street_number'];
+        $address_query['street_number_suffix'] = $contact_data['street_number_suffix'];
```

Comment:
Do we always have `street_number_suffix` if there's `street_number`? Or is another existence check required?

Reply:
Yes I do think so. But just to be sure I have added an if statement.